core/src/mongosqltranslate.rs

```diff
@@ -42,36 +31,60 @@ fn get_library_name(library_type: &str) -> String {
     }
 }
 
-fn get_library_path(library_type: &str) -> PathBuf {
+#[cfg(not(test))]
+fn get_library_path(library_type: &str) -> Result<PathBuf> {
     let lib_name = get_library_name(library_type);
-    let mut path = PathBuf::from(LIBRARY_INSTALL_PATH);
-    path.push(lib_name);
-    path
+
+    let settings = DriverSettings::from_private_profile_string()
+        .map_err(|e| Error::LibraryPathError(format!("Failed to obtain driver settings: {}", e)))?;
+
+    let driver_path = PathBuf::from(settings.driver);
+    let library_dir = driver_path.parent().ok_or_else(|| {
+        Error::LibraryPathError(format!(
+            "Failed to get parent directory from driver path: '{}'",
+            driver_path.display()
+        ))
+    })?;
+
+    let path = PathBuf::from(library_dir).join(lib_name);
+    Ok(path)
 }
 
-fn get_mock_library_path() -> PathBuf {
-    let lib_name = get_library_name(MOCK_LIBRARY_NAME);
-    let mut path = env::current_exe().unwrap();
-    path.pop();
-    path.pop();
-    path.push("deps"); // Go to the 'deps' directory where the library should be located
-    path.push(lib_name);
-    path
+#[cfg(test)]
+fn get_library_path(library_type: &str) -> Result<PathBuf> {
+    let lib_name = if cfg!(target_os = "windows") {
+        format!("mock_{}.{}", library_type, LIBRARY_EXTENSION)
+    } else {
+        format!("libmock_{}.{}", library_type, LIBRARY_EXTENSION)
+    };
+
+    let exe_path = env::current_exe().map_err(|e| {
+        Error::LibraryPathError(format!("Failed to get current executable path: {}", e))
+    })?;
+    let exe_dir = exe_path.parent().ok_or_else(|| {
+        Error::LibraryPathError("Failed to get executable's parent directory".to_string())
+    })?;
+
+    let path = PathBuf::from(exe_dir).join(lib_name);
+    Ok(path)
 }
 
-// load_mongosqltranslate_library is the entry point for loading the mongosqltranslate library
+// load_mongosqltranslate_library is the entry point for loading the mongosqltranslate library.
+// The library is expected to be in the same directory as the MongoDB ODBC driver.
 // and is responsible for determining the library name and path.
 // The library name and path are determined based on the operating system and architecture.
 // It is stored in a static variable to ensure that it is only loaded once.
```

Comment:
```suggestion
// load_mongosqltranslate_library is the entry point for loading the mongosqltranslate library
// and is responsible for determining the library name and path.
// The library name and path are determined based on the operating system and architecture.
// Additionally, the library is expected to be in the same directory as the MongoDB ODBC driver.
// The library is stored in a static variable to ensure that it is only loaded once.
```
I think this makes it a little more clear.